File: Day14/QueueImplementation/QueueImplement.py
class QueueImp:
    def __init__(self):
        self.arr=[]
    
    def isEmpty(self):
        if(len(self.arr)==0):
            return True
        else:
            return False
        
    def enqueue(self,num):
        self.arr.append(num)

    def dequeue(self):
        if(self.isEmpty()):
            print("We cant remove any element ")
        else:
            result=self.arr.pop(0)
            print("pop element=",result)
    
    def display(self):
        for i in (self.arr):
            print(i,end=" ")
        print("")
    
    def peek(self):
        print("Top element:=",self.arr[0])

    def size(self):
        print("Size of stack=",len(self.arr))


obj= QueueImp()
obj.enqueue(12)
obj.enqueue(23)
obj.enqueue(34)
print("Queue element:")
obj.display()
obj.dequeue()
obj.dequeue()
print("Queue element:")
obj.display() 

obj.peek()
obj.size()

File: Day14/QueueImplementation/QueueImplement.java
package Day14.QueueImplementation;

class Queue
{
    int arr[];
    int front; 
    int rear;
    public Queue()
    {
        arr=new int[5];
        front=rear=-1;
    }
    public void enqueue(int num)
    {
        if(isFull())
        {
            System.out.println("Queue is full");
        }
        else
        {
            if(front==-1&&rear==-1)
            {
                front++;
                rear++;
                arr[rear]=num;
            }
            else
            {
                rear++;
                arr[rear]=num;
            }
        }
    }
    public boolean isFull()
    {
        if(rear==arr.length-1)
        {
            return true;
        }
        else
            return false;
    }
    public boolean isEmpty()
    {
        if(rear==-1)
        {
            return true;
        }
        else
            return false;
    }
    public void dequeue()
    {
        if(isEmpty())
            System.out.println("Queue is empty");
        else
        {
            System.out.println("Dequeue element:="+arr[front]);
            front++;
        }

    }

    public void display()
    {
        for(int i=front;i<=rear;i++)
        {
            System.out.print(arr[i]+" ");
        }
        System.out.println("");
    }
}

public class QueueImplement {
    public static void main(String[] args) {
        Queue obj=new Queue();
        obj.enqueue(10);
        obj.enqueue(23);
        obj.enqueue(54);
        obj.enqueue(50);
        obj.enqueue(100);
        System.out.println("Queue Element");
        obj.display();
        obj.enqueue(78);
        obj.dequeue();
        System.out.println("Queue Element");
        obj.display();
        obj.enqueue(78);
        System.out.println("Queue Element");
        obj.display();
       
    }
}
